python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -385,25 +386,32 @@ def generate_session_token(hostname, login, password, http_proxy):
     :param login: The user to get a session for.
     :param password: Password for the user.
     :param http_proxy: Proxy to use. Can be None.
+    :param auth_token: Two factor authentication token for the user. Can be None.
 
-    :returns: The generated session token for that user/password/site combo.
+    :returns: The generated session token for that user/password/site combo. If the login
```

Comment:
Alright, looking at all the invocation points for generate_session_token, it does seem odd to single out two factor authentication errors (N) and handle it differently than the others.

Especially since if we had some form of three factor authentication (N+1) in the future, I would be back at throwing exceptions out of the method since there would be no reasonable way to return a new error code from the method.

There, proof by induction. I knew that would come handy at some point. :)